// package Basic_Sorting;

public class BubbleSort {
    public static void print(int arr[]){
        for(int ele: arr){
            System.out.print(ele+" ");
        }
        System.out.println();

    }
    public static void main(String[] args) {
        int arr[]={5,-2,6,7,2,0,7,2};
        int n=arr.length;
        print(arr);

        // for(int i=0; i<=n-1; i++){ // n-1 passes
        //     for(int j=0; j<n-1-i; j++){
        //         if(arr[j]>arr[j+1]){
        //             int temp=arr[j];
        //             arr[j]=arr[j+1];
        //             arr[j+1]=temp;
        //         }
        //     }
        // }


        // // Bubble sort better
        // for(int i=0; i<n-1; i++){
        //     boolean isSorted= true;
        //     for(int j=0; j<n-1-i; j++){
        //         if(arr[j]>arr[j+1]){
        //             int temp=arr[i];
        //             arr[j]=arr[j+1];
        //             arr[j+1]=temp;
        //             isSorted=false;
        //         }
        //     }
        //     if(isSorted==true) break

        // }

        // // Alternate way to do.. 
        for(int i=0;i<n-1;i++){
            int swap=0;
            for(int j=0;j<n-1-i;j++){
                if(arr[j]>arr[j+1]){
                    int temp=arr[j];
                    arr[j]=arr[j+1];
                    arr[j+1]=temp;
                    swap++;
                }
            }
            if(swap==0) break;
        }
        print(arr);

    }
}
